src/robot.cpp

```diff
@@ -64,22 +63,30 @@ auto Robot::from(std::shared_ptr<moveit::core::RobotModel const> const& model,
     return robot;
 }
 
-auto Robot::get_random_valid_configuration() const -> std::vector<double> {
-    std::vector<double> config;
+auto Robot::set_random_valid_configuration(std::vector<double>& config) const -> void {
     auto const num_vars = variables.size();
-    config.reserve(num_vars);
+    if (config.size() != num_vars) {
+        config.resize(num_vars);
+    }
     for (size_t idx = 0; idx < num_vars; ++idx) {
-        auto const var = variables[idx];
-        config.push_back(rsl::uniform_real(var.clip_min, var.clip_max));
+        auto const& var = variables[idx];
+        if (var.bounded) {
```

Comment:
Kind of related to the comment above re: functions in `Variable` structs. There are a lot of if/elses, clamping, and this kind random number generations that might be easier of they were just wrapped up in the struct's interface. You could always add something like `var.get_uniform_real()` that just handle things like this for you?

No strong opinions here, just something I noticed.